doc/rolling_update.md

```diff
@@ -0,0 +1,283 @@
+to be used by name in all plays:
+
+```yaml
+# hosts.yml
+
+all:
+  vars:
+    ...
+    cartridge_custom_scenarios:
+      update_and_restart_instance:
+        - update_instance
+        - restart_instance
+        - wait_instance_started
+    ...
+```
+
+**Note** that [`update_instance` step](/doc/scenario.md#update_instance) requires
+`cartridge_package_path` variable to set instance application link to unpacked
+distribution.
+
+The example rolling update playbook:
+
+```yaml
+# playbook.yml
+
+- name: Deliver and install myapp 2.0.0
+  hosts: all
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario:
+      - deliver_package
+      - update_package
+    cartridge_package_path: ./myapp-2.0.0.tar.gz
+
+- name: Update stateboard application version
+  hosts: "my-stateboard"
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario_name: update_and_restart_instance
+    cartridge_package_path: ./myapp-2.0.0.tar.gz
+
+- name: Update storages replicas application version
+  hosts: "*storage*replica*"
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario_name: update_and_restart_instance
+    cartridge_package_path: ./myapp-2.0.0.tar.gz
+
+- name: Promote storages leaders to replicas
+  hosts: all
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario:
+      - failover_promote
+    cartridge_failover_promote_params:
+      replicasets_leaders:
+        storage-1: storage-1-replica
+        storage-2: storage-2-replica
+
+- name: Update storages leaders application version
+  hosts: "*storage*leader*"
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario_name: update_and_restart_instance
+    cartridge_package_path: ./myapp-2.0.0.tar.gz
+
+- name: Promote storages leaders back
+  hosts: all
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario:
+      - failover_promote
+    cartridge_failover_promote_params:
+      replicasets_leaders:
+        storage-1: storage-1-leader
+        storage-2: storage-2-leader
+
+- name: Update routers application version
+  hosts: "*core*"
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario_name: update_and_restart_instance
+    cartridge_package_path: ./myapp-2.0.0.tar.gz
+
+- name: Remove old packages
+  hosts: all
+  roles:
+    - tarantool.cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario:
+      - rotate_dists
+    cartridge_keep_num_latest_dists: 1
+```
+
+## Rolling update: Playbook for RPM or DEB
```

Comment:
What about to describe what is "RPM" or "DEB"? Maybe "playbook for deploying RPM or DEB packages".

Reply:
See at the top of file:

```
Here is described rolling update for an application that uses
TGZ package with [multiversion approach](/doc/multiversion.md) or
usual RPM/DEB packages.
```

Reply:
```suggestion
## Rolling update: Playbook for deploying RPM or DEB packages
```